### PR TITLE
[mds-metrics] Adds geography_id and provider_id query params to /metr…

### PR DESCRIPTION
**Pull request by macsj200**
…ics/all

Usage: GET /metrics/all?geography_id={}&provider_id={}

Both fields are optional.
Each id query param must be a valid UUID or you'll get a 500.

TODO test w/ cURL or Insomnia @evanxadkins

This is also going into PR #88

Closes [MDSAAS-487](https://lacunadotai.atlassian.net/browse/MDSAAS-487)

## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

**Comment by avatarneil** [comment]
This `geography` filtering will currently _never_ work. The processors do not insert a `geography_id` in the payload. Furthermore, there isn't even a `geography_id` in a `MetricsTableRow`... there is a `geography` however.

File: packages/mds-metrics/request-handlers.ts [hunk 1]
```diff
@@ -198,8 +198,15 @@ export async function getEventCounts(req: MetricsApiRequest, res: GetEventCounts
...
 */
 export async function getAll(req: MetricsApiRequest, res: GetAllResponse) {
-  const { body } = req
+  const { body, query } = req
   const slices = getTimeBins(body)
+  const geography_id = query.geography_id || null
+  const provider_id = query.provider_id || null
+
+  if (geography_id !== null && !isUUID(geography_id))
+    return res.status(400).send(new ServerError(`geography_id ${geography_id} is not a UUID`))
```
Comment (avatarneil) [inline packages/mds-metrics/request-handlers.ts]: ServerError isn't the proper error type to use here, should be a `BadParamsError`

Comment (drtyh2o) [reply packages/mds-metrics/request-handlers.ts]: `BadParamsError` and `ValidationError` are largely similar and should probably be consolidated.

**Comment by macsj200** [comment]
That is unfortunate, see discussion here https://lacuna-tech.slack.com/archives/CPY98QSS3/p1575564328001100

**Comment by macsj200** [comment]
Goodness gracious this PR got out of hand quickly! Sorry about that to the reviewers.

In the interest of expediency I'd rather just get it all approved in one go, but I'm certainly amenable to the notion of splitting it up if the reviews prove too onerous.

Here's an updated provisional **tl;dr**:

- Adds optional `format` query param (defaults to `'json'`)
- Adds relative date/time parsing logic
- Adds `vehicle_type` querying support

...among other things

Of the many CSV (TSV?) npm packages, I chose[json2csv](https://www.npmjs.com/package/json2csv) because it seems to receive the most love and has a relatively straightforward API. If you'd like to see some sample output, see [this test case](https://github.com/lacuna-tech/mds-core/pull/98/files#diff-f51b638a487f409e55f2df37fbcb2753R111). Based on the type definition of `MetricsTableRow`, the JSON should serialize readily to TSV rows.

File: packages/mds-utils/utils.ts [hunk 2]
```diff
@@ -689,6 +693,150 @@ const calcDistance = (telemetry: TripTelemetry[][], startGps: GpsData) => {
...
+  if (offset === 'now') {
+    return localTime
+  }
+  if (offset === 'yesterday') {
+    return localTime.startOf('day').subtract(1, 'days')
+  }
+  throw new BadParamsError(`Invalid anchor point: ${offset}`)
+}
+
+// For details on time parameters and ailases, see https://docs.google.com/document/d/1Zyn58tHo-VzibsdgmFU3fBcDlplUDxFzGWLHtTG4-Cs/edit?pli=1#bookmark=id.5snf8ffk8bjf
```
Comment (avatarneil) [inline packages/mds-utils/utils.ts]: Linking to an internal g-doc is no bueno for FOSS ;)

**Pull request merged**